### ISSUE TITLE
python profiler doesn't seem to work any more

### ISSUE THREAD

**Issue by robmcmullen** [robmcmullen/peppy#435]
_[Trac time 20080429 210453Z]_ Something is wrong with the profiler -- trying to run peppy with the "-p" option results in:

```
Exception exceptions.AssertionError: AssertionError('Bad call', ('/home/rob/src/peppy-git/peppy/main.py', 313, 'OnInit'), <frame object at 0x17c6760>, <
frame object at 0x1735050>, <frame object at 0x1732440>, <frame object at 0x1825e80>) in <bound method Font.<lambda> of <wx._gdi.Font;  >> ignored
```

which seems to occur right after the call to

```
         GlobalPrefs.convertConfig()
```

in the main OnInit method.

**Comment by robmcmullen** [comment]
_[Trac time 20090521 004457Z]_ [old svn rev2046] Refs #435: Fixed regression introduced by bbe91a77eafb4d44c896f224969ba27abfeabd067
- if found another peppy instance, should exit by returning from the main function rather than app.Exit() -- the main loop hasn't been entered yet!

**Comment by robmcmullen** [comment]
_[Trac time 20090520 185259Z]_ Found http://bugs.python.org/issue5978 which may be my problem: sys.exit() doesn't allow the profiler to finish up its statistics.  Should be using wx.App.Exit() instead.

**Comment by robmcmullen** [comment]
_[Trac time 20090520 192211Z]_ Found an excellent profile visualizer here: http://www.vrplumber.com/programming/runsnakerun/

**Comment by robmcmullen** [comment]
_[Trac time 20090520 232935Z]_ [old svn rev2037] Fixed #435: replaced sys.exit() calls with wx.GetApp().Exit() calls
- sys.exit didn't allow the profiler to write the statistics
- replaced profile with cProfile
